subsys/zigbee/common/zigbee_ota_client.c

```diff
@@ -0,0 +1,490 @@
+	} else {
+		app_image_ver.major = mcuboot_header.h.v1.sem_ver.major;
+		app_image_ver.minor = mcuboot_header.h.v1.sem_ver.minor;
+		app_image_ver.revision =  mcuboot_header.h.v1.sem_ver.revision;
+	}
+
+	/* Basic cluster attributes data */
+	dev_ctx.basic_attr.zcl_version  = ZB_ZCL_VERSION;
+	dev_ctx.basic_attr.power_source = ZB_ZCL_BASIC_POWER_SOURCE_UNKNOWN;
+
+	/* OTA cluster attributes data */
+	zb_ieee_addr_t addr = ZB_ZCL_OTA_UPGRADE_SERVER_DEF_VALUE;
+
+	ZB_MEMCPY(dev_ctx.ota_attr.upgrade_server, addr,
+		  sizeof(zb_ieee_addr_t));
+	dev_ctx.ota_attr.file_offset = ZB_ZCL_OTA_UPGRADE_FILE_OFFSET_DEF_VALUE;
+	dev_ctx.ota_attr.file_version = app_image_ver.uint32_ver;
+	dev_ctx.ota_attr.stack_version =
+		ZB_ZCL_OTA_UPGRADE_FILE_HEADER_STACK_PRO;
+	dev_ctx.ota_attr.downloaded_file_ver  =
+		ZB_ZCL_OTA_UPGRADE_DOWNLOADED_FILE_VERSION_DEF_VALUE;
+	dev_ctx.ota_attr.downloaded_stack_ver =
+		ZB_ZCL_OTA_UPGRADE_DOWNLOADED_STACK_DEF_VALUE;
+	dev_ctx.ota_attr.image_status =
+		ZB_ZCL_OTA_UPGRADE_IMAGE_STATUS_DEF_VALUE;
+	dev_ctx.ota_attr.manufacturer = CONFIG_ZIGBEE_OTA_MANUFACTURER_ID;
+	dev_ctx.ota_attr.image_type = CONFIG_ZIGBEE_OTA_IMAGE_TYPE;
+	dev_ctx.ota_attr.min_block_reque = 0;
+	dev_ctx.ota_attr.image_stamp = ZB_ZCL_OTA_UPGRADE_IMAGE_STAMP_MIN_VALUE;
+}
+
+
+uint32_t zb_ota_get_image_ver(void)
+{
+	return app_image_ver.uint32_ver;
+}
+
+void zb_ota_dfu_init(void)
+{
+	memset(&ota_ctx, 0, sizeof(ota_ctx));
+}
+
+void zb_ota_dfu_abort(void)
+{
+	LOG_INF("ABORT Zigbee DFU");
+	/* Reset the counters */
+	ota_ctx.total_size = 0;
+	ota_ctx.ota_header_fill_level = 0;
+	dfu_target_done(false);
+}
+
+/* Code for rebooting the chip */
+void zb_ota_reboot_application(zb_uint8_t param)
+{
+	ARG_UNUSED(param);
+
+	LOG_INF("Reboot application.");
+	int res = dfu_target_reset();
+
+	if (res != 0) {
+		LOG_ERR("Unable to reset DFU target");
+	}
+	sys_reboot(SYS_REBOOT_COLD);;
+}
+
+
+void zb_ota_client_init(void)
+{
+	/* Marks the currently running image as confirmed.*/
+	if (!boot_is_img_confirmed()) {
+		int ret = boot_write_img_confirmed();
+		if (ret) {
+			LOG_ERR("Couldn't confirm image: %d", ret);
+		} else {
+			LOG_INF("Marked image as OK");
+		}
+	}
+
+	ota_client_attr_init();
+
+	/* Initialize periodic OTA server discovery */
+	zb_ota_client_periodical_discovery_srv_init();
+}
+
+void zb_ota_confirm_image(void)
+{
+	if (!boot_is_img_confirmed()) {
+		int ret = boot_write_img_confirmed();
+		if (ret) {
+			LOG_ERR("Couldn't confirm image: %d", ret);
+		} else {
+			LOG_INF("Marked image as OK");
+		}
+	}
+}
+
+static void dfu_target_callback_handler(enum dfu_target_evt_id evt)
+{
+	switch (evt) {
+	case DFU_TARGET_EVT_TIMEOUT:
+		LOG_INF("DFU_TARGET_EVT_TIMEOUT");
+		break;
+	case DFU_TARGET_EVT_ERASE_DONE:
+		LOG_INF("DFU_TARGET_EVT_ERASE_DONE");
+		break;
+	default:
+		LOG_INF("DFU_TARGET_EVT_ERROR");
+	}
+}
+
+static uint8_t ota_clt_dfu_target_init(const uint8_t *magic_word_buf)
+{
+	int err = 0;
+
```

Comment:
If `_clt_` is a reference to `client`, what about changing it to `_cli_`?

Reply:
Yes, it is a reference to client.